File: AndroidServiceCommunication/app/src/main/java/app/serviceconnector/activities/ServiceSelectionActivity.java
package app.serviceconnector.activities;

import android.app.ListActivity;
import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.AdapterView;
import android.widget.ArrayAdapter;
import android.widget.ListView;

import app.serviceconnector.services.DemoBoundForgroundService;
import app.serviceconnector.services.DemoBoundIntentService;
import app.serviceconnector.services.DemoBoundService;

public final class ServiceSelectionActivity extends ListActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        ListView lv = getListView();

        lv.setAdapter(new ArrayAdapter<>(getApplication(), android.R.layout.simple_list_item_1, new String[]{
                "Bound Service Demo", "Bound Foreground Service Demo", "Bound Intent Service Demo"
        }));

        lv.setOnItemClickListener(new AdapterView.OnItemClickListener() {
            @Override
            public void onItemClick(AdapterView<?> parent, View view, int position, long id) {
                String service = "";
                switch (position) {
                    case 0:
                        service = DemoBoundService.class.getName();
                        break;
                    case 1:
                        service = DemoBoundForgroundService.class.getName();
                        break;
                    case 2:
                        service = DemoBoundIntentService.class.getName();
                        break;
                }
                Intent intent = new Intent(getApplication(), ServiceLauncherActivity.class);
                intent.putExtra("SERVICE", service);
                startActivity(intent);
                finish();
            }
        });
    }
}

File: AndroidServiceCommunication/app/src/main/java/app/serviceconnector/activities/ServiceSelectionActivity.kt
package app.serviceconnector.activities

import android.app.ListActivity
import android.content.Intent
import android.os.Bundle
import android.widget.AdapterView
import android.widget.ArrayAdapter
import app.serviceconnector.services.DemoBoundForegroundService
import app.serviceconnector.services.DemoBoundIntentService
import app.serviceconnector.services.DemoBoundService

class ServiceSelectionActivity : ListActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        listView.adapter = ArrayAdapter(
            application,
            android.R.layout.simple_list_item_1,
            arrayOf(
                "Bound Service Demo",
                "Bound Foreground Service Demo",
                "Bound Intent Service Demo"
            )
        )

        listView.onItemClickListener =
            AdapterView.OnItemClickListener { parent, view, position, id ->
                var service = ""
                when (position) {
                    0 -> service = DemoBoundService::class.java.name
                    1 -> service = DemoBoundForegroundService::class.java.name
                    2 -> service = DemoBoundIntentService::class.java.name
                }
                val intent = Intent(application, ServiceLauncherActivity::class.java)
                intent.putExtra("SERVICE", service)
                startActivity(intent)
                //finish()
            }
    }
}
